samples/cellular/lwm2m_client/src/main.c

```diff
@@ -385,15 +385,23 @@ static void rd_client_event(struct lwm2m_ctx *client, enum lwm2m_rd_client_event
 {
 	k_mutex_lock(&lte_mutex, K_FOREVER);
 
+	if (client_state == LTE_OFFLINE && client_event != LWM2M_RD_CLIENT_EVENT_ENGINE_SUSPENDED) {
+		LOG_DBG("Drop network event %d at LTE offline state", client_event);
+		k_mutex_unlock(&lte_mutex);
+		return;
+	}
+
 	if (IS_ENABLED(CONFIG_LWM2M_CLIENT_UTILS_LTE_CONNEVAL)) {
 		lwm2m_utils_conneval(client, &client_event);
 	}
 
-	if (client_state == LTE_OFFLINE &&
-	    client_event != LWM2M_RD_CLIENT_EVENT_ENGINE_SUSPENDED) {
-		LOG_DBG("Drop network event %d at LTE offline state", client_event);
-		k_mutex_unlock(&lte_mutex);
-		return;
+	if (IS_ENABLED(CONFIG_LWM2M_CLIENT_UTILS_RAI) &&
+	    client_event == LWM2M_RD_CLIENT_EVENT_QUEUE_MODE_RX_OFF) {
+		lwm2m_rai_last();
+	}
+
+	if (IS_ENABLED(CONFIG_LWM2M_CLIENT_UTILS_DTLS_CON_MANAGEMENT)) {
```

Comment:
Move this `IS_ENABLED()` into the function itself.

The main application is becoming too messy, so rather have all checks in the utility library.